### PR TITLE
store: fix data race about KVStore.tikvClient

### PR DESCRIPTION
**Pull request by Yisaer**
Signed-off-by: yisaer <disxiaofei@163.com>

### What problem does this PR solve?

Issue Number: close #xxx 

Problem Summary:

https://github.com/pingcap/tidb/issues/24586

As KVStore have an extra goroutine to update safeTS which will read its tikvClient, there are other tests may call `SetTiKVClient` in other goroutines. This causes data race.

### What is changed and how it works?
use atomic for client

### Check List 

Tests 

- Unit test

### Release note 

* No release note

### CODE REVIEW

File: store/tikv/kv.go [hunk 1]
```diff
@@ -66,10 +66,13 @@ var oracleUpdateInterval = 2000
...
 // KVStore contains methods to interact with a TiKV cluster.
 type KVStore struct {
-	clusterID    uint64
-	uuid         string
-	oracle       oracle.Oracle
-	client       Client
+	clusterID uint64
+	uuid      string
+	oracle    oracle.Oracle
+	clientMu  struct {
```
Comment (JmPotato) [inline store/tikv/kv.go:72]: What about using `atomic.Value`?

Comment (Yisaer) [reply store/tikv/kv.go]: updated.

**Pull request merged**